test/api/controllers/remoteActionsController/swagger.test.js

```diff
@@ -0,0 +1,113 @@
+var
+  rc = require('rc'),
+  fs = require('fs'),
+  params = rc('kuzzle'),
+  q = require('q'),
+  rewire = require('rewire'),
+  should = require('should'),
+  sinon = require('sinon'),
+  Kuzzle = require.main.require('lib/api/Kuzzle'),
+  InternalError = require.main.require('kuzzle-common-objects').Errors.internalError,
+  swagger = require.main.require('lib/api/controllers/remoteActions/swagger');
+
+require('sinon-as-promised')(q.Promise);
+
+describe('Test: Swagger files generation', () => {
+  var
+    kuzzle,
+    sandbox;
+
+  before(() => {
+    kuzzle = new Kuzzle();
+
+    return kuzzle.start(params, {dummy: true});
+  });
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+
+    sandbox.stub(kuzzle.config, 'httpRoutes', [
+      {verb: 'get', url: '/basic', controller: 'basic', action: 'basic'},
+      {verb: 'post', url: '/complete', controller: 'complete', action: 'complete', infos: {description: 'description'}}
+    ]);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it ('should create the swagger files when called', () => {
+    sandbox.stub(fs, 'writeFileSync', () => {
+      return true;
+    });
+
+    swagger(kuzzle)
+      .then(() => {
+        should(fs.writeFileSync.calledTwice).be.true();
+      });
+  });
+
+  it ('should create entries in the swagger files for the plugins', () => {
+    sandbox.stub(fs, 'writeFileSync', () => {
+      return true;
+    });
+
+    sandbox.stub(kuzzle.pluginsManager, 'routes', [
+      {verb: 'post', url: '/myplugin/bar', controller: 'myplugin/foo', action: 'bar'},
+      {verb: 'get', url: '/myplugin/bar/:name', controller: 'myplugin/foo', action: 'bar'}
+    ]);
+
+    swagger(kuzzle)
```

Comment:
When testing promises with should, you either need a `done` callback or, better yet, to return that promise, ensuring that `should` will wait for it to be fulfilled/rejected before marking the test as a success/failure